packages/x-charts/src/BarChart/BarPlot.tsx

```diff
@@ -157,8 +157,8 @@ const useAggregatedData = (): {
             layout,
             x: verticalLayout ? xScale(baseValue)! + barOffset : minValueCoord,
             y: verticalLayout ? minValueCoord : yScale(baseValue)! + barOffset,
-            xOrigin: xScale(0)!,
-            yOrigin: yScale(0)!,
+            xOrigin: xScale(0) ?? 0,
+            yOrigin: yScale(0) ?? 0,
             height: verticalLayout ? maxValueCoord - minValueCoord : barWidth,
```

Comment:
Not sure how to handle this. Technically, defaulting this to 0 is wrong, but it's also wrong to assert as non-undefined because it can be. 

To have proper types, I suppose we'd need to change the return value depending on the layout. 

Open to suggestions.

Reply:
```
const xOrigin = xScale(0)

if (!xOrigin) throw 💣 
```

🤷 

I think defaulting to 0 is reasonable enough, even though it could create issues, it shouldn't happen

Reply:
If we throw our tests will start failing. The thing is that we don't use the value in the cases where it doesn't make sense, so there's no issue. If we throw, we'll break our tests.